### ISSUE TITLE
Proxyjump passphrase error

### ISSUE THREAD

**Issue by wmegchel** [ronf/asyncssh#413]
Hi Ron, 

I use asyncssh to connect through a bastion server to a server on a protected domain. I get the following error:

```bash
Traceback (most recent call last):
  File "async.py", line 49, in <module>
    conn = asyncio.get_event_loop().run_until_complete(Connect.create(host, user, passphrase))
  File "C:\Users\wmegc\miniconda3\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File "async.py", line 32, in create
    self.conn = await asyncssh.connect(host=host, username=user , passphrase=passphrase) # , client_keys=['C:/Users/wmegc/.ssh/id_rsa_hpc'])
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 6855, in connect
    return await _connect(options, loop, flags, conn_factory,
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 274, in _connect
    new_tunnel = await _open_tunnel(tunnel)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 258, in _open_tunnel
    return await connect(host, port, username=username)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 6850, in connect
    options = SSHClientConnectionOptions(options, config=config, host=host,
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 5596, in __init__
    super().__init__(options=options, last_config=last_config, **kwargs)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\misc.py", line 268, in __init__
    self.prepare(**self.kwargs)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\connection.py", line 6307, in prepare
    self.client_keys = load_keypairs(client_keys, passphrase,
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 3153, in load_keypairs
    key, certs = read_private_key_and_certs(key, passphrase)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 2970, in read_private_key_and_certs
    key, cert = import_private_key_and_certs(read_file(filename), passphrase)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 2860, in import_private_key_and_certs
    key, end = _decode_private(data, passphrase)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 2520, in _decode_private
    key = _decode_pem_private(pem_name, headers, data, passphrase)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 2434, in _decode_pem_private
    return _decode_openssh_private(data, passphrase)
  File "C:\Users\wmegc\miniconda3\lib\site-packages\asyncssh\public_key.py", line 2277, in _decode_openssh_private
    raise KeyImportError('Passphrase must be specified to import '
asyncssh.public_key.KeyImportError: Passphrase must be specified to import encrypted private keys
```

I am sure the password is set and correct. Instead, the error was introduced when I modified my ~/.ssh/config and changed this line:

```bash
# This works
Proxycommand ssh -xaqW%h:22 bastion_server
```

To this
```
# This does not work
ProxyJump bastion_server
```

From the command line, this works fine. With asyncssh I get the error above.  When I modified `public_key.py` to print the passphrase it receives. In the first case (proxycommand), it prints:
```
>>>>> DEBUG: function _decode_openssh_private received password: secret_pass <<<<
``` 
using the ProxyJump it receives the password twice, and one of them is empty:

```
>>>>> DEBUG: function _decode_openssh_private received password: secret_pass <<<<
>>>>> DEBUG: function _decode_openssh_private received password: None <<<<
```

No big deal, but hope you can fix it for the next release.

Best,
Wout

**Comment by ronf** [comment]
Thanks for the report! I think I see the problem. The options you pass in to the original connect() call are not being provided when AsyncSSH opens the tunnel connection. So, it would only pick up things set in your SSH config file, which by design doesn't let you specify a passphrase.

This is fixable, but it's a bit tricky given that tunnels can be used for both outbound and inbound connections, and the options passed for setting up a listener for inbound connections can not be passed directly when opening the outbound tunnel connection. I have a few thoughts around how I can work around that with a separate tunnel_options argument that could default to the original connection options in the common case where both connections are outbound but I'll need a little time to see how to most cleanly add that.

In the meantime, if you want to use AsyncSSH for both connections, you could open the connection to the bastion host explicitly and pass the SSHClientConnection which gets returned as a "tunnel" argument on a second connect() to the final destination host. That'll let you set whatever arguments you need in each of the two connect() calls. In this case, you'll want to remove the ProxyJump/ProxyCommand from the config file.

**Comment by ronf** [comment]
This is now available in the "develop" branch as commit 16becd8.

**Comment by ronf** [comment]
There were some problems with this commit, so I ended up reworking it as commit 4976307, which is now available in release 2.8.1. If you have any further problems with this, feel free to re-open this issue or create a new one.
